### DISCUSSION TITLE
How should root's path be modified by a formula?

### DISCUSSION THREAD

**Discussion by gdiamond** [Homebrew/discussions#2231] [Writing Formulae/Casks]
I installed `mtr` today on macOS Catalina but it was not in root's or my user's path when the installation was completed. 
I understand that `mtr` is an admin application installed in `/usr/local/sbin`.  It's not meant to be used by any old user so it doesn't need to be in my path but it should be in root's path.  But when I try running as suggested I get `sudo: mtr: command not found`.

```
==> Homebrew was updated to version 3.2.14
The changelog can be found at:
  https://github.com/Homebrew/brew/releases/tag/3.2.14
==> Downloading https://ghcr.io/v2/homebrew/core/mtr/manifests/0.94
######################################################################## 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/mtr/blobs/sha256:6ec962809c1c2381b7723647e16c2283dbef8042ae04af14ad675fa63c38a859
==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:6ec962809c1c2381b7723647e16c2283dbef8042ae04af14ad675fa63c38a859?se=2021-10-01T18%3A20%3A00Z&sig=ZdqqFJtBAdwFoPYqObFs8uJB9aIZm3ZAl5BuW7Y%2F%2F4w%3D&sp=r&spr=https&sr=b&sv=2019-12-12
######################################################################## 100.0%
==> Pouring mtr--0.94.catalina.bottle.tar.gz
==> Caveats
mtr requires root privileges so you will need to run 'sudo mtr'.
You should be certain that you trust any software you grant root privileges.
==> Summary
🍺  /usr/local/Cellar/mtr/0.94: 12 files, 255.5KB

 % sudo mtr 9.9.9.9
Password:
sudo: mtr: command not found
```
Does this happen for everyone?

I was able to fix the problem by creating `/etc/paths.d/mtr` with the path `/usr/local/sbin`.
```
sudo echo "/usr/local/sbin" >> /etc/paths.d/mtr
```
Is this an appropriate solution?  Maybe the formula can suggest to the user to run the command if they want it added to the path?   Otherwise, the recommendation to `run 'sudo mtr'` should be changed to `run 'sudo /usr/local/sbin/mtr'`.

I haven't seen other brew formulas doing it this way but some installers are doing it.
```
-rwxr-xr-x   1 root  wheel    23 Apr  2  2021 100-rvictl
-rw-r--r--   1 root  wheel    43 Oct 22  2020 Wireshark
-r--r--r--   1 root  wheel    47 Jun  1 12:20 com.vmware.fusion.public
```

**Comment by gromgit** [comment]
> [`/usr/local/sbin`] should be in root's path

It is:
```
$ sudo -i
Password:
# echo $PATH
/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Users/aho/bin:/Users/aho/bin/Darwin-x86_64:/usr/local/sbin
```

The problem you're seeing is that `sudo` looks up and runs the command you give it by default, so it's using your current `PATH`. From the `sudo` man page:
> `-i`, `--login`  Run the shell specified by the target user's password database entry as a login shell.  This means that login-specific resource files such as `.profile`, `.bash_profile` or `.login` will be read by the shell.  If a command is specified, it is passed to the shell for execution via the shell's `-c` option.  If no command is specified, an interactive shell is executed.

So try running this instead:
```
$ sudo -i mtr 9.9.9.9
```

**Reply by gdiamond** [reply]
I understand the `/usr/local/bin` **should** be in root's path.  This is not so in my environment which is a factory installation of Catalina.  Hence my question and my intention of helping improve the formula.

Is there an acceptable way of modifying root's path?  Is there an acceptable way of checking root's path?  Or is the more appropriate way to add a helpful note at the end of installation?
